components/Provider/AuthProvider.tsx

```diff
@@ -104,10 +116,12 @@ export const AuthProvider = ({ children }: Props): JSX.Element => {
     () => ({
       currentUser,
       currentUserBalance,
+      atomicMarketBalance,
       authError,
       login,
       logout,
       updateCurrentUserBalance,
+      updateAtomicBalance,
     }),
     [currentUser, authError, currentUserBalance]
```

Comment:
We need to add `atomicMarketBalance` to the dependency array, otherwise the components won't re-render whenever `atomicMarketBalance` is updated.